frontend/app/controllers/digital_objects_controller.rb

```diff
@@ -69,6 +71,20 @@ def new
       end
     end
 
+    if user_prefs['digital_object_spawn_description']
+      if params[:spawn_from_resource_id]
+        copy_from_record = Resource.find(params[:spawn_from_resource_id])
+      elsif params[:spawn_from_accession_id]
+        copy_from_record = Accession.find(params[:spawn_from_accession_id])
+      elsif params[:spawn_from_archival_object_id]
+        copy_from_record = ArchivalObject.find(params[:spawn_from_accession_id])
```

Comment:
Apologies! I think I managed to fool myself into believing this was working because when you refresh on editing an archival_object in the resource tree, it goes back up to the top level resource, and in my haste I did not realize I was testing the wrong thing. 😬 There were a couple more changes required.